pkg/controller/storagecluster/storagecluster.go

```diff
@@ -209,6 +214,15 @@ func (c *Controller) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, err
 	}
 
+	if c.waitingForMigrationApproval(cluster) {
```

Comment:
this should probably be after the validation, so users know if the validation fails or not.

Reply:
That's a good point. One of the validations that we have marks the storagecluster as failed due to storagecluster being present. I will make changes to it in the subsequent PR where we do actual migration. 
Once that is done, I will move this check after the validation.